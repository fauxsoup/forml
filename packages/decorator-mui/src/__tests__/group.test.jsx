import Group from "../group";
import Context from "@forml/context";
import React from "react";
import { render } from "@testing-library/react";
import * as decorator from "../";

describe("renders", function() {
    let form;
    let title = "title";
    let description = "description";

    beforeEach(function() {
        form = { type: "string" };
    });

    describe("with form options", function() {
        let fields = {
            fullWidth: [true, false],
        };

        Object.keys(fields).forEach(function(field) {
            fields[field].forEach(function(value) {
                describe(`${field}`, function() {
                    test(`${value}`, function() {
                        form = { ...form, [field]: value };
                        const { container } = render(
                            <Context.Provider value={{ decorator }}>
                                <Group
                                    form={form}
                                    title={title}
                                    description={description}
                                />
                            </Context.Provider>
                        );

                        expect(container).toMatchSnapshot();
                    });
                });
            });
        });
    });

    test("with no title or description", function() {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <Group form={form} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
