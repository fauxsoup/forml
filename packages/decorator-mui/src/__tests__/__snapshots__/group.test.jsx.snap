// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with form options fullWidth false 1`] = `
<div>
  <div
    class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
  />
</div>
`;

exports[`renders with form options fullWidth true 1`] = `
<div>
  <div
    class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
  />
</div>
`;

exports[`renders with no title or description 1`] = `
<div>
  <div
    class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
  />
</div>
`;
